apps/examples/simple_file_transfer/sft_main.c

```diff
@@ -0,0 +1,200 @@
+		} else if (nbytes < 0) {
+			printf("unknown error(%d)\n", errno);
+			return -1;
+		}
+		read_size += nbytes;
+		*readed = read_size;
+	}
+	return 0;
+}
+
+static int tcp_server_thread(void)
+{
+	struct sockaddr_in servaddr;
+	struct sockaddr_in cliaddr;
+	int listenfd = -1;
+	int connfd = -1;
+	socklen_t clilen;
+	int ret = 0;
+	int readed = 0;
+
+	listenfd = socket(PF_INET, SOCK_STREAM, 0);
+	if (listenfd < 0) {
+		printf("[TCPSERV] TCP socket failure %d\n", errno);
+		return -1;
+	}
+
+	/*
+	 * Enable SO_REUSEADDR to allow multiple instances of this
+	 * application to receive copies of the multicast datagrams.
+	 */
+
+	int reuse = 1;
+	ret = setsockopt(listenfd, SOL_SOCKET, SO_REUSEADDR, (char *)&reuse, sizeof(reuse));
+	if (ret < 0) {
+		printf("[TCPSERV] ERR: setting SO_REUSEADDR\n");
+		goto out_with_socket;
+	}
+	printf("[TCPSERV] set reusable success\n");
+
+	/* Connect the socket to the server */
+	memset(&servaddr, 0, sizeof(servaddr));
+	servaddr.sin_family = PF_INET;
+	servaddr.sin_addr.s_addr = htonl(INADDR_ANY);
+	servaddr.sin_port = htons(SFT_PORT);
+
+	ret = bind(listenfd, (struct sockaddr *)&servaddr, sizeof(servaddr));
+	if (ret < 0) {
+		perror("[TCPSERV] bind fail\n");
+		goto out_with_socket;
+	}
+
+	printf("[TCPSERV] Listening... port %d\n", SFT_PORT);
+
+	ret = listen(listenfd, 1024);
+	if (ret < 0) {
+		perror("[TCPSERV] listen fail\n");
+		goto out_with_socket;
+	}
+
+	clilen = sizeof(cliaddr);
+
+	connfd = accept(listenfd, (struct sockaddr *)&cliaddr, &clilen);
+	if (connfd < 0) {
+		perror("[TCPSERV] accept fail\n");
+		goto out_with_socket;
+	}
+	printf("[TCPSERV] Accepted\n");
+
+	int file_name_size = 0;
+	ret = read_data(connfd, (char *)&file_name_size, sizeof(int), &readed);
+	if (ret < 0) {
+		printf("readed %d %d\n", readed, __LINE__);
+		close(connfd);
+		goto out_with_socket;
+	}
+	file_name_size = ntohl(file_name_size);
+	printf("file name length(%d) %d\n", file_name_size, readed);
+	char *tmp_name;
+	tmp_name = (char *)&file_name_size;
+	printf("%d %d %d %d\n", tmp_name[0], tmp_name[1], tmp_name[2], tmp_name[3]);
+
+	ret = read_data(connfd, g_name_buf, file_name_size, &readed);
+	if (ret < 0) {
+		printf("fail %d\n", __LINE__);
+		close(connfd);
+		goto out_with_socket;
+	}
+	g_name_buf[file_name_size] = 0;
+	printf("file name: %s\n", g_name_buf);
+
+	int file_size = 0;
+	ret = read_data(connfd, (char *)&file_size, sizeof(int), &readed);
+	if (ret < 0) {
+		printf("error %d\n", __LINE__);
+		close(connfd);
+		goto out_with_socket;
+	}
+	file_size = ntohl(file_size);
+	printf("file size = %d\n", file_size);
+
+	FILE *fp = fopen(g_name_buf, "w");
+	if (!fp) {
+		printf("fail %d\n", __LINE__);
+		close(connfd);
+		goto out_with_socket;
+	}
+
+	int remain = file_size;
+	while (remain > 0) {
+		readed = 0;
+		int buf_size = remain < BUF_SIZE ? remain : BUF_SIZE;
+		ret = read_data(connfd, g_buf, buf_size, &readed);
+		if (readed > 0) {
+			fwrite(g_buf, sizeof(char), readed, fp);
+		}
+		if (ret < 0) {
```

Comment:
1. it should be moved to above of ' if (readed > 0)', in my opinion.
2. return of fwrite also need to be considered. if it didn't write same as readed, extra logic is needed.